src/kademlia/node.cpp

```diff
@@ -497,8 +508,20 @@ struct ping_observer : observer
 				node_id id;
 				std::copy(nodes, nodes + 20, id.begin());
 				nodes += 20;
-				algorithm()->get_node().m_table.heard_about(id
-					, detail::read_v4_endpoint<udp::endpoint>(nodes));
+#if TORRENT_USE_IPV6
+				if (at == ipv4)
+#endif
+				{
+					algorithm()->get_node().m_table.heard_about(id
+						, detail::read_v4_endpoint<udp::endpoint>(nodes));
```

Comment:
how about just reading the endpoint into a generic udp::endpoint in the if-block, and then just have a single call to heard_about()?